### DISCUSSION TITLE
MSI Motherboard fails to load ZFSBootMenu

### DISCUSSION THREAD

**Discussion by gmemstr** [zbm-dev/zfsbootmenu#672] [General]
### ZFSBootMenu build source

Release EFI

### ZFSBootMenu version

2.3.0

### Boot environment distribution

Gentoo Linux

### Problem description

Less of a ZFSBootMenu issue and more of a MSI motherboard issue (specifically the [MAG B650 TOMAHAWK WIFI
](https://www.msi.com/Motherboard/MAG-B650-TOMAHAWK-WIFI/support)), but thought it worthwhile to report here while I investigate further.

Attempting to use the latest firmware available for the motherboard results in it hanging when directed to boot ZFSBootMenu. I have yet to narrow down the exact version that introduced the issue, but it's somewhere between `7D75v15` and `7D75v1J`. No logs are produced that could help narrow down the issue.

Not expecting maintainers to fix this, but making a note so if anything else hits this they have a place to land from Google (hopefully). When I have time I plan to track down the exact version and raise an issue with MSI directly.

### Steps to reproduce

1. Flash latest version of BIOS available for an MSI motherboard
2. Attempt to boot ZFSBootMenu either with auto-detected UEFI OS entry or efibootmgr-added entry

**Comment by jonhermansen** [comment]
FYI, I'm running ZFSBootMenu just fine on an MSI PRO Z690-A DDR4 mainboard.

**Comment by arborealoctopus** [comment]
same issue, followed void instructions on a dedicated drive -- msi only sees windows

i know this has been a problem for the void install instructions in the past. Im going to try to glue together the zfs instructions with https://docs.voidlinux.org/installation/guides/chroot.html#installing-on-removable-media-or-non-compliant-uefi-systems and see how far I get.

**Reply by classabbyamp** [reply]
you can put the ZBM efi at the path `EFI/BOOT/BOOTX64.EFI` within the ESP (which is the fallback path) or use a tool like rEFInd to launch ZBM

**Reply by arborealoctopus** [reply]
Awesome thank you, that worked for me (i just went the copy route, I think once the motherboard recognized one it saw the others as well). If it helps anyone else, the refind version (I just did plain zbm but tried refind first) seems to be settable to the compatibility location with a flag per the arch wiki: `refind-install --usedefault /dev/sdXY`
